src/forms/renderers/Overrides/material/complex/MaterialOneOfRenderer_Format.tsx

```diff
@@ -0,0 +1,207 @@
+    const handleClose = useCallback(() => setOpen(false), [setOpen]);
+    const cancel = useCallback(() => {
+        setOpen(false);
+    }, [setOpen]);
+
+    const oneOfRenderInfos = createCombinatorRenderInfos(
+        (schema as JsonSchema).oneOf as JsonSchema[],
+        rootSchema,
+        'oneOf',
+        uischema,
+        path,
+        uischemas
+    );
+
+    // Run through the elements and clear out the ones without elements
+    oneOfRenderInfos.map((renderer) => {
+        const { uischema: rendererUischema } = renderer as any;
+        rendererUischema.elements = rendererUischema.elements.filter(
+            (el: any) => el !== null
+        );
+        return renderer;
+    });
+
+    const openNewTab = (newIndex: number) => {
+        const tabSchema = oneOfRenderInfos[newIndex].schema;
+        const { properties: tabSchemaProps } = tabSchema;
+        const defaultVal: {
+            formatter: string;
+            settings?: any;
+        } = {
+            formatter: tabSchemaProps?.formatter.default ?? '',
+        };
+
+        if (has(tabSchemaProps, 'settings')) {
+            defaultVal.settings = {};
+        }
+
+        handleChange(path, defaultVal);
+        setSelectedIndex(newIndex);
+    };
+
+    const confirm = useCallback(() => {
+        handleChange(path, null);
+        openNewTab(newSelectedIndex);
+        setOpen(false);
+    }, [handleChange, createDefaultValue, newSelectedIndex]);
+
+    const handleTabChange = useCallback(
+        (_event: any, newOneOfIndex: number) => {
+            setNewSelectedIndex(newOneOfIndex);
+            if (isEmpty(data) || !has(data, 'settings')) {
+                openNewTab(newOneOfIndex);
+            } else {
+                setOpen(true);
+            }
+        },
+        [setOpen, setSelectedIndex, data]
+    );
+
+    return (
+        <Hidden xsUp={!visible}>
+            <CombinatorProperties
+                schema={schema}
+                combinatorKeyword="oneOf"
+                path={path}
+            />
+            <Tabs value={selectedIndex} onChange={handleTabChange}>
+                {oneOfRenderInfos.map((oneOfRenderInfo) => (
+                    <Tab
+                        key={oneOfRenderInfo.label}
+                        label={oneOfRenderInfo.label}
+                    />
+                ))}
+            </Tabs>
+            {oneOfRenderInfos.map(
+                (oneOfRenderInfo, oneOfIndex) =>
+                    selectedIndex === oneOfIndex && (
+                        <JsonFormsDispatch
+                            key={oneOfIndex}
+                            schema={oneOfRenderInfo.schema}
+                            uischema={oneOfRenderInfo.uischema}
+                            path={path}
+                            renderers={renderers}
+                            cells={cells}
+                        />
+                    )
+            )}
+            <Dialog
```

Comment:
All text in this dialog should be placed in the language file.

Reply:
I'm going to leave it here since it isn't actually "our content". This needs to stay in sync with JSON Forms and I don't want folks thinking they can just change.

I'm gonna add this note to the ReadMe so it is documented in the code.